hr_data_analytics_package_folder/api.py

```diff
@@ -0,0 +1,165 @@
+# Basic libraries
+import pandas as pd
+import json
+
+#Pipelines and transformers
+from sklearn.pipeline import make_pipeline
+
+# Models
+from sklearn.ensemble import GradientBoostingClassifier
+from xgboost import XGBClassifier
+
+# Import from .py files
+from hr_data_analytics_package_folder.ml_logic.data import clean_data_hiring, clean_data_leaving
+from hr_data_analytics_package_folder.ml_logic.preprocessor import preprocess_features_hiring, preprocess_features_leaving
+from hr_data_analytics_package_folder.params import *
+
+from fastapi import FastAPI, UploadFile
+
+app = FastAPI()
+
+@app.get('/')
+def index():
+    return {'HR data analytics project': 'This is the first app of our project !!!'}
+
+@app.post("/upload_predict_hiring")
+async def create_upload_file(file: UploadFile):
+    X_pred = await file.read(size=-1)
+    X_pred = X_pred.decode().replace("'", '"')
+    X_pred = json.loads(X_pred)
+    # X_pred = pd.DataFrame(X_pred, index =[0])
+# ************************************
+    # At least 2 applicants !!!
+    new_dict = {}
```

Comment:
Hi Thierry, why do we need to transform a JSON file (dictionnary) into a new dictionnary to then transform it into a dataframe instead of transforming it straight into a dataframe ?

Reply:
Hello Louis ! I use 'file.read(size=-1)' to access the content of file. It returns a byte file. That's why I had to clean it before use. 
But I am still looking for a method to get the content of 'file' directly as a json (as we upload a json file). No tricks/cleaning would be necessary. Still researching though.
We can keep this topic open. Thanks for the feedback.